src/components/composer/content_editable_helper.ts

```diff
@@ -157,4 +195,29 @@ export class ContentEditableHelper {
       endSelectionOffset: selection.focusOffset,
     };
   }
+
+  getText(): string {
+    let text = "";
+
+    let it = iterateChildren(this.el);
+    let current = it.next();
+    let isFirstParagraph = true;
+    while (!current.done) {
```

Comment:
I feel like I do 3 time almost the same loop, there's probably a way to improve that. Not something I can do on a friday afternoon tho

Reply:
`.innerText` don't work very well with the current impelmentation with `<p>` in the composer : 
- a newLine is returned as TWO `\n` characters, or three if the next line is an empty line
- if the last line is empty with a span with the `selectionIndicator` class, it's totally ignored in `.innerText`

I could solve these issues but : 
- I'm afraid of missing other edges cases
- I'm afraid that these could be browser specific

So not a fan of using `.innerText` with the current impelemntation. Will try it again if I manage to replace the `<p>` by spans with `<br>`